go/oasis-node/cmd/common/common.go

```diff
@@ -191,12 +193,21 @@ func GetInputReader(cmd *cobra.Command, cfg string) (io.ReadCloser, bool, error)
 }
 
 // LoadEntity loads the entity and it's signer.
-func LoadEntity(dataDir string) (*entity.Entity, signature.Signer, error) {
+func LoadEntity(backing string) (*entity.Entity, signature.Signer, error) {
 	if flags.DebugTestEntity() {
 		return entity.TestEntity()
 	}
 
 	// TODO/hsm: Configure factory dynamically.
-	entitySignerFactory := fileSigner.NewFactory(dataDir, signature.SignerEntity)
-	return entity.Load(dataDir, entitySignerFactory)
+	switch backing {
+	case ledgerSigner.SignerName:
+		factory := ledgerSigner.NewFactory(flags.EntityFileDir(), signature.SignerEntity)
+		signer, err := factory.Load(signature.SignerEntity)
+		return nil, signer, err
```

Comment:
Please explain why the entity is not loaded and nil is returned here? It should be exactly the same invocation as for the file based signer (`entity.Load(DataDir(), factory)`)?

Reply:
Maybe a misunderstanding on my part, but I didn't think there would necessarily be an Entity associated with the Ledger device. If there is always an associated Entity, then I guess the dataDir is always required.